PacketDotNet/OSPFv2Packet.cs

```diff
@@ -0,0 +1,1056 @@
+        ///</summary>
+        /// See <see cref="PacketDotNet.LSA"/>
+        public virtual List<LSA> LSAHeader
+        {
+            get
+            {
+                List<LSA> ret = new List<LSA>();
+                int bytesNeeded = this.PacketLength - OSPFv2Fields.LSAHeaderPosition;
+
+                if (bytesNeeded % OSPFv2Fields.LSAHeaderLength != 0)
+                {
+                    throw new Exception("OSPFv2 DD Packet - Invalid LSA headers count");
+                }
+
+                int offset = this.header.Offset + OSPFv2Fields.LSAHeaderPosition;
+                int headerCount = bytesNeeded / OSPFv2Fields.LSAHeaderLength;
+
+                for (int i = 0; i < headerCount; i++)
+                {
+                    LSA l = new LSA(this.header.Bytes, offset , OSPFv2Fields.LSAHeaderLength);
+                    offset += OSPFv2Fields.LSAHeaderLength;
+                    ret.Add(l);
+                }
+                return ret;
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder packet = new StringBuilder();
+            packet.Append(base.ToString());
+            packet.Append(" ");
+            packet.AppendFormat("DDSequence: 0x{0:X8} ", this.DDSequence);
+            packet.AppendFormat("#LSA headers: {0} ", this.LSAHeader.Count);
+            return packet.ToString();
+        }
+
+        public override string ToString(StringOutputType outputFormat)
+        {
+            return ToString();
+        }
+    }
+
+    /// <summary>
+    /// Link State Request packets are OSPF packet type 3.
+    /// The Link State Request packet is used to request the pieces of the
+    /// neighbor's database that are more up-to-date.
+    /// See http://www.ietf.org/rfc/rfc2328.txt for details.
+    /// </summary>
+    public class OSPFv2LSRequestPacket : OSPFv2Packet
+    {
+        /// <value>
+        /// The packet type
+        /// </value>
+        public static OSPFPacketType packetType = OSPFPacketType.LinkStateRequest;
+
+        /// <summary>
+        /// Constructs an OSPFv2 LSR packet
+        /// </summary>
+        public OSPFv2LSRequestPacket()
+        {
+            this.Type = packetType;
+            this.PacketLength = (ushort)header.Bytes.Length;
+        }
+
+        /// <summary>
+        /// Constructs an OSPFv2 LSR packet with link state requests
+        /// </summary>
+        /// <param name="lsrs">List of the link state requests</param>
+        public OSPFv2LSRequestPacket(List<LinkStateRequest> lsrs)
+        {
+            int length = lsrs.Count * LinkStateRequest.Length;
+            int offset = OSPFv2Fields.HeaderLength;
+            byte[] bytes = new byte[length + OSPFv2Fields.HeaderLength];
+
+            Array.Copy(header.Bytes, bytes, header.Length);
+            for (int i = 0; i < lsrs.Count; i++)
+            {
+                Array.Copy(lsrs[i].Bytes, 0, bytes, offset, LinkStateRequest.Length); //20 bytes per header

```

Comment:
It may make sense to forgo mentioning 20 bytes per header in the comment here if the user can look at the actual value of LinkStateRequest.Length. I don't feel particularly strong about it but the thinking is that you've duplicated information in the comment and the value itself and if one of them changes you'll have them out of sync.